### DISCUSSION TITLE
Support for generic types

### DISCUSSION THREAD

**Discussion by vaishnavmhetre** [Netflix/dgs-framework#956] [Q&A]
Is there someway to hack create a generic FilterField input type to support different types for value? Like String, Int, Uuid, etc.
Schema just needs to check out valid, even if it is a hack and not have newly curated functional implementation as I'm yet experimenting with dgs.

Just to support, I've found a transpiler [GraphqlS2S Transpiler - Generic Types](https://github.com/nicolasdao/graphql-s2s#generic-types) supporting js based impls.

I'm not able to find the link to resource but somewhere I saw implementation having `@generic` directive on extended/implemented types and its definitions.

Can someone help me achieve this using some inheritance overriding possibility? Or should I just have some Java level mappings with Generic Comparator impl?

**Comment by jord1e** [comment]
> Just to support, I've found a transpiler [GraphqlS2S Transpiler - Generic Types](https://github.com/nicolasdao/graphql-s2s#generic-types) supporting js based impls.

This is not part of the GraphQL standard, and not a part of GraphQL Java parsing/

> Just to support, I've found a transpiler [GraphqlS2S Transpiler - Generic Types](https://github.com/nicolasdao/graphql-s2s#generic-types) supporting js based impls.

This transpiler first parses the schema itself, then converts the AST to something GraphQL.js can use, *in practice you could generate the schema and then feed it into GraphQL Java (DGS) by just saving it.*

> I'm not able to find the link to resource but somewhere I saw implementation having @generic directive on extended/implemented types and its definitions.

This is not a part of DGS (or the normal GraphQL specification)

> Can someone help me achieve this using some inheritance overriding possibility? Or should I just have some Java level mappings with Generic Comparator impl?
I can't figure out what this means, why not just write your schema normally instead of adding complexity?

Short answer: No.

**Reply by vaishnavmhetre** [reply]
The specification for the same is in discussion due to enormous places of such requirements!

[Raised ticket at Graphql-Spec](https://github.com/graphql/graphql-spec/issues/190)

**Comment by jord1e** [comment]
Yes, but it is not in the specification

The transpiler is exactly what it says it is, it transpiles non-official syntax to official syntax by generating types.

This is not implemented by GraphQL Java nor DGS, you will have to fork GraphQL Java and add support to the parser.

You can write the types on your own, it really is not that much work x)

**Reply by vaishnavmhetre** [reply]
Okay so can we consider this for a possibile future scope in DGS as Java natively supports Generics?

**Reply by jord1e** [reply]
The generics support from java does not transfer to the graphql language, they are not comparable in any way shape or form.

It will only be supported if the graphql spec decides to add generics (it probably will not)

I will reiterate:
> You can write the types on your own, it really is not that much work x)

**Reply by vaishnavmhetre** [reply]
Okay lets see if the raised ticket for generics support passes as it was opened in graphql-spec if you take a look in previous comment!

[Support for Generics - Graphql-Spec](https://github.com/graphql/graphql-spec/issues/190)

**Reply by jord1e** [reply]
It has not received any attention recently, and the proposal has been open since 2016

https://github.com/graphql/graphql-wg/search?q=generics

Don't get your hopes up :)
